site/zenodo_rdm/legacy/serializers/schemas/legacyjson.py

```diff
@@ -139,22 +139,22 @@ def dump_files(self, obj):
         result = []
         files_url = obj["links"].get("files")
         for key, f in obj["files"].get("entries", {}).items():
-            file_id = f["id"]
-            if files_url:
+            file_id = f.get("id")
+            if file_id is not None:  # Check if 'id' is present
```

Comment:
Maybe add a comment why an id is not there when file entries are? i.e the problem with pending files.